### PR TITLE
Implemented code injection

### PR DESCRIPTION
**Pull request by rohitsanj**
Implements #11 

- Created a basic implementation of code injection. 
- renamed `notebook` fixture to `notebook_loader`

It works as follows:
```python
from testbook import notebook_loader

def foo():
   print('I ran in foo')

def test_foo():
   with notebook_loader('/path/to/notebook') as notebook:
      notebook.inject(foo).assert_output_text('I ran in foo')
```

And a function can be injected when called with arguments as well,
```python
from testbook import notebook_loader

def foo_with_args(arg1, arg2):
   print(f'You passed {arg1} and {arg2}')

def test_foo_with_args():
   with notebook_loader('/path/to/notebook') as notebook:
      notebook.inject(foo_with_args, ['value1', 'value2']).assert_output_text('You passed value1 and value2')
```

cc @MSeal

### CODE REVIEW

**Comment by rohitsanj** [comment]
I've placed comments and used arbitrary variable names in some places. This will need a refactor later on.


TODO:
- Add a case where the entire code block is passed as a string.

File: testbook/testbooknode.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+class TestbookNode(NotebookNode):
+    """
+    Extends `NodebookNode` to perform assertions
+    """
+
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+
+    def assert_output_text(self, expected_text):
```
Comment (MSeal) [inline testbook/testbooknode.py:12]: Nice!

File: testbook/client.py [hunk 2]
```diff
@@ -48,3 +56,38 @@ def cell_output(self, cell_index):
         """
 
         outputs = self.nb['cells'][cell_index]['outputs']
+
+    def inject(self, func, args=None):
```
Comment (MSeal) [inline testbook/client.py:60]: I think we need to specify what cell to inject this before or after as kwargs. We likely need to be able to reference local variables executed in earlier cells. However, I'm also fine with merging this as is and tackling local variable reference in a later PR.

Comment (rohitsanj) [reply testbook/client.py:60]: Sure, I think we can look at that in a later PR.

Also, we could include an optional argument called `prerun` which can pre-run a list of cells. Something like, 

```python
def test_foo_with_args(notebook_loader):
   with notebook_loader('/path/to/notebook') as notebook:
      notebook.inject(foo_with_args, ['value1', 'value2'], prerun=[1,3,4]).assert_output_text('You passed value1 and value2')
```

**Comment by rohitsanj** [comment]
> Well done, we can either merge now or expand it a little and then merge.

Thanks! I think we can merge this now.

**Pull request merged**